### PR TITLE
Fix for `trailing_semicolon` rule wrongly also removing trailing comments

### LINKED ISSUE #4817: Auto-fixing `trailing_semicolon` rule deletes trailing comments
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When SwiftLint auto-fixes a `trailing_semicolon` violation, it also deletes any comment after the semicolon. It shouldn't.

#### Original line:
```
private let pulseVelocity: CGFloat = 878 / 2;  // 878 px / 2 seconds
```

#### Command:
```bash
$ swiftlint lint --fix
```

#### Expected:
```
private let pulseVelocity: CGFloat = 878 / 2  // 878 px / 2 seconds
```

#### Actual:
```
private let pulseVelocity: CGFloat = 878 / 2
```

### Environment

* 0.50.3
* Homebrew
* Paste your configuration file:

```yml
disabled_rules:
 - identifier_name
 - todo
 - trailing_comma
 - type_body_length
 - function_body_length

included:
 - Encircle

excluded:
 - Pods

# Rule configurations
force_cast: warning

class_delegate_protocol: error

cyclomatic_complexity:
  error: 34

identifier_name:
  min_length:
    error: 0
    warning: 3
  excluded:
    - id

line_length: 120
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcodebuild -version`)? 14.2

### PR DESCRIPTION
**Pull request by mildm8nnered**
Resolves #4817

When replacing the semi-colon, preserve the trailing trivia.

Ironically, the original comment read 

`// Is there a better way to remove a node? Should we somehow keep trailing trivia?`

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Idiomatic/TrailingSemicolonRule.swift [hunk 1]
```diff
@@ -65,8 +66,8 @@ private extension TrailingSemicolonRule {
             }
 
             correctionPositions.append(node.positionAfterSkippingLeadingTrivia)
-            // Is there a better way to remove a node? Should we somehow keep trailing trivia?
-            return super.visit(TokenSyntax(.semicolon, presence: .missing))
+            // Keep the trailing trivia
+            return ("" as TokenSyntax).with(\.trailingTrivia, node.trailingTrivia ?? .zero)
```
Comment (mildm8nnered) [inline Source/SwiftLintFramework/Rules/Idiomatic/TrailingSemicolonRule.swift:70]: Not sure If this is the best way to do this

Comment (SimplyDanny) [reply Source/SwiftLintFramework/Rules/Idiomatic/TrailingSemicolonRule.swift:69]: Not sure either. This seems to boil down to `.identifier("")` which is a bit strange. Do you like `.unknown("").with(\.trailingTrivia, ...)` more?

**Review by SimplyDanny — APPROVED** [review]

File: Source/SwiftLintFramework/Rules/Idiomatic/TrailingSemicolonRule.swift (hunk 1 above)
Comment (mildm8nnered) [reply Source/SwiftLintFramework/Rules/Idiomatic/TrailingSemicolonRule.swift]: I think so - it feels less hacky than the cast, so I went with that.

**Comment by SimplyDanny** [comment]
Thank you for the prompt fix!

**Pull request merged**